### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by poorva1209**
This change is [Reviewable](https://reviewable.io/reviews/volttron/volttron/854)

### CODE REVIEW

File: docs/source/devguides/walkthroughs/Agent-Authentication-Walkthrough.rst [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+::
+
+   @RPC.export
+   @RPC.allow('can_call_bar')
+   def bar(self):
+      return 'If you can see this, then you have the required capabilities'
+
+
+AgentB's atuhentication record to access bar method:
+
```
Comment (craig8) [inline docs/source/devguides/walkthroughs/Agent-Authentication-Walkthrough.rst]: Spelling

File: docs/source/devguides/walkthroughs/Agent-Authentication-Walkthrough.rst [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+NOTE: Currently VOLTTRON uses only CURVE mechanism to authenticate agents.
+
+::
+
+    mechanism [NULL]: CURVE
+
+Credentials:
+-------------
+
```
Comment (craig8) [inline docs/source/devguides/walkthroughs/Agent-Authentication-Walkthrough.rst]: #Verify with @zbeech if plain can be used.

File: docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+Assumptions
+-----------
+
+-  Device driver agent is already developed
+
+Configuration
+-------------
+
+1. Copy example configuration file applications/pnnl/DrivenMatlabAgent/config\_waterheater to volltron/config.
```
Comment (craig8) [inline docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst]: Note that this isn't in the same tree.

Comment (craig8) [inline docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst]: Probably needs to be after cloning applications

File: docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+   '{"commands":{"device1":[["point1",value1]],"device2":[["point2",value2]]}}';
+
+Steps to test integration:
+--------------------------
+
+1. Install VOLTTRON on a VM or different system than the one
+   running Matlab.
+
+    Follow link: http://volttron.readthedocs.io/en/develop/install.html
+
```
Comment (craig8) [inline docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst]: This isn't the proper way to do this

File: volttron/platform/syslog.py [hunk 5]
```diff
@@ -0,0 +1,142 @@
+# -*- coding: utf-8 -*- {{{
+# vim: set fenc=utf-8 ft=python sw=4 ts=4 sts=4 et:
+
+# Copyright (c) 2015, Battelle Memorial Institute
+# All rights reserved.
```
Comment (craig8) [inline volttron/platform/syslog.py]: Booo this shouldn't be here

File: docs/source/devguides/walkthroughs/Agent-Authentication-Walkthrough.rst (hunk 1 above)
Comment (poorva1209) [inline docs/source/devguides/walkthroughs/Agent-Authentication-Walkthrough.rst]: Spelling corrected.

File: docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst (hunk 3 above)
Comment (poorva1209) [inline docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst]: DrivenMatlabAgent is under applications tree.

File: docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst (hunk 4 above)
Comment (poorva1209) [inline docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst]: what do you mean by this ?

File: volttron/platform/syslog.py (hunk 5 above)
Comment (poorva1209) [inline volttron/platform/syslog.py]: Removed

File: docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst (hunk 3 above)
Comment (craig8) [inline docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst]: I was talking about the application waterheater thing

File: docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst (hunk 4 above)
Comment (craig8) [inline docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst]: Yeah this needs to be a sub-repository.  The format for the clone is a bit different I believe.

File: docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst (hunk 3 above)
Comment (poorva1209) [inline docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst]: Aah! got it ...changing it to after cloning.

File: docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst (hunk 4 above)
Comment (poorva1209) [inline docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst]: Didn't get it. What needs to be a sub-repo ? VOLTTRON installation ?

File: docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst (hunk 3 above)
Comment (poorva1209) [inline docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst]: Done.

File: docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst (hunk 4 above)
Comment (craig8) [inline docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst]: http://blogs.atlassian.com/2013/05/alternatives-to-git-submodule-git-subtree/

Sorry not sub-repo subtree.

Comment (poorva1209) [inline docs/source/devguides/walkthroughs/DrivenMatlabAgent-Walkthrough.rst]: Updated subtree command as: 
git subtree add --prefix applications https://github.com/VOLTTRON/volttron-applications.git develop --squash

Done. Please review.

**Pull request merged**